sysstat.service.in

```diff
@@ -2,20 +2,27 @@
 # (C) 2012 Peter Schiffer (pschiffe <at> redhat.com)
 #
 # @PACKAGE_NAME@-@PACKAGE_VERSION@ systemd unit file:
-#	 Insert a dummy record in current daily data file.
-#	 This indicates that the counters have restarted from 0.
+#     1. Insert a dummy record in current daily data file when starting/stopping
+#        the unit. This indicates that the counters have restarted from 0.
+#     2. This unit is automatically stopped/started on suspend/resume,
+#        see /usr/lib/systemd/system-sleep/sysstat-sleep.sh.
+#     3. Starting this unit also starts the timer unit for periodic accounting.
 
 [Unit]
-Description=Resets System Activity Logs
+Description=System Activity Accounting
+# Start the periodic accounting process
+Requires=sysstat-collect.timer
+# but make sure we finish first, because the accounting process locks the datafile
+Before=sysstat-collect.timer sysstat-collect.service
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
 User=@CRON_OWNER@
-ExecStart=@SA_LIB_DIR@/sa1 --boot
+ExecStart=@SA_LIB_DIR@/sa1 --boot -C STARTED
```

Comment:
> You can always be sure that if the counter is lower than the previous value, then a reboot has happened.

True.
> If the counter is higher than the previous value, then yes, you can't be sure if a reboot has happened. But you are /almost/ positive it hasn't. Because If the service is properly enabled and started on every boot then sadc should be called no sooner than when it was called after the previous boot, in the vast majority of cases.

You're right again. But anyway I need a way to know that a reboot happened that works all the time for all the possible situations, and inserting the LINUX_RESTART message is that way to do it. Be sure that if you don't handle all the cases, you will always have somebody out there who will end filling a bug report and complaining that "it doesn't work". :-(

Wrt auto-detection of system reboot, the logic could be based on the number of jiffies spent in the different modes (user, nice, etc.) by cpu "all" (this is the first line from /proc/stat). If one of those counters has decreased then a reboot happened. But please try to avoid that solution since it won't work all the time as I have just told you.